templates/shared/user/profile_big_avatar.tmpl

```diff
@@ -99,12 +99,12 @@
 			{{if and .IsSigned (ne .SignedUserID .ContextUser.ID)}}
 			<li class="follow">
 				{{if $.IsFollowing}}
-					<form method="post" action="{{.Link}}?action=unfollow&redirect_to={{$.Link}}">
+					<form method="post" action="{{.ContextUser.HomeLink}}?action=unfollow&redirect_to={{$.Link}}">
 						{{$.CsrfTokenHtml}}
 						<button type="submit" class="ui basic red button">{{svg "octicon-person"}} {{.locale.Tr "user.unfollow"}}</button>
 					</form>
 				{{else}}
-					<form method="post" action="{{.Link}}?action=follow&redirect_to={{$.Link}}">
+					<form method="post" action="{{.ContextUser.HomeLink}}?action=follow&redirect_to={{$.Link}}">
```

Comment:
```suggestion
					<form method="post" action="{{.ContextUser.HomeLink}}?action=follow&redirect_to={{$.Link}}&tab={{.TabName}}">
```

Reply:
Add this to `user/profile.Action` at the end.
```go
redirectTo := ctx.FormString("redirect_to")
tab := ctx.FormString("tab")
if tab != "" {
redirectTo = fmt.Sprintf("%s?tab=%s", redirectTo, tab)
}
// FIXME: We should check this URL and make sure that it's a valid Gitea URL
ctx.RedirectToFirst(redirectTo, ctx.ContextUser.HomeLink())
```

Reply:
See above, this approach is too hacky, I don't want to make it more complex.